packages/core/src/domain/trackingConsent.ts

```diff
@@ -0,0 +1,38 @@
+import { ExperimentalFeature, isExperimentalFeatureEnabled } from '../tools/experimentalFeatures'
+import { Observable } from '../tools/observable'
+
+export const TrackingConsent = {
+  GRANTED: 'granted',
+  NOT_GRANTED: 'not-granted',
+} as const
+export type TrackingConsent = (typeof TrackingConsent)[keyof typeof TrackingConsent]
+
+export interface TrackingConsentState {
+  setIfNotDefined: (trackingConsent: TrackingConsent) => void
+  set: (trackingConsent: TrackingConsent) => void
+  isGranted: () => boolean
+  observable: Observable<void>
+}
+
+export function createTrackingConsentState(currentConsent?: TrackingConsent): TrackingConsentState {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** To be consistent with our namings I would expect to have‏
```suggestion
export function startTrackingConsent(currentConsent?: TrackingConsent): TrackingConsentState {
```

Reply:
Mmh we have a bunch of `createXXX` function as well, I kind of prefer sticking to the `create` naming when returning the associated type.